pipeline/api/cloud.py

```diff
@@ -55,6 +54,24 @@
 FILE_CHUNK_SIZE = 200 * 1024 * 1024  # 200 MiB
 
 
+class CallbackBytesIO(io.BytesIO):
+    """Provides same interface as BytesIO but additionally calls a callback function
+    whenever the 'read' method is called.
+
+    This is similar to tqdm's own CallbackIOWrapper but this does not play nicely with
+    all features of httpx so we use our own in some cases.
+    """
+
+    def __init__(self, callback: Callable, initial_bytes: bytes):
+        self._callback = callback
+        super().__init__(initial_bytes)
+
+    def read(self, size=-1) -> bytes:
+        data = super().read(size)
+        self._callback(len(data))
+        return data
```

Comment:
not sure if worth moving this into the `util` folder?

Reply:
moved